### PR TITLE
Add support for koa

### PR DESCRIPTION
**Pull request by timokoessler**
Adds support for https://www.npmjs.com/package/koa
Used by Strapi, for example.
Requires new hook system #324

### CODE REVIEW

File: library/middleware/koa.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { shouldBlockRequest } from "./shouldBlockRequest";
+import { escapeHTML } from "../helpers/escapeHTML";
+import type Application = require("koa");
```
Comment (hansott) [inline library/middleware/koa.ts:3]: not possible with

`import type * as Application from "koa"`? :D

File: library/sinks/Path.test.ts [hunk 2]
```diff
@@ -109,9 +109,7 @@ t.test("it works", async (t) => {
     const error = t.throws(() => join("/etc/", "test.txt"));
     t.same(
       error instanceof Error ? error.message : null,
-      getMajorNodeVersion() <= 22
-        ? "Zen has blocked a path traversal attack: path.normalize(...) originating from body.file.matches"
-        : "Zen has blocked a path traversal attack: path.join(...) originating from body.file.matches"
+      "Zen has blocked a path traversal attack: path.normalize(...) originating from body.file.matches"
```
Comment (hansott) [inline library/sinks/Path.test.ts:112]: Why is this changed?

File: library/agent/protect.ts [hunk 3]
```diff
@@ -132,6 +134,8 @@ function getWrappers() {
     new BetterSQLite3(),
     new Postgresjs(),
     new Fastify(),
+    new Koa(),
+    new KoaRouter(),
```
Comment (hansott) [inline library/agent/protect.ts:138]: I wouldn't mind these being one sink but up to you!

File: library/sources/koa/wrapMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable max-lines-per-function */
```
Comment (hansott) [inline library/sources/koa/wrapMiddleware.ts:1]:
```suggestion

```

File: library/sinks/Path.test.ts (hunk 2 above)
Comment (timokoessler) [reply library/sinks/Path.test.ts:112]: Was changed in Node.js v21.1 😄

File: library/middleware/koa.ts (hunk 1 above)
Comment (timokoessler) [reply library/middleware/koa.ts:3]: Ahh, yes 😅. Typescript always recommended this when trying to import with `import type { Application } from "koa"`.

File: library/agent/protect.ts (hunk 3 above)
Comment (timokoessler) [reply library/agent/protect.ts:138]: Because the koa router already consists of two different packages to wrap, it's maybe a bit better readable if it's splitted. Maybe there are more koa routers to add in the future.

**Review by hansott — APPROVED** [review]
